File: P32_gbeex_v4/gbeex-backend/models/subjects.py
# gbeex-backend/models/subjects.py
from datetime import datetime
from typing import List, Optional, Dict, Any
from pydantic import BaseModel, Field

class VitalSigns(BaseModel):
    bloodPressure: str
    heartRate: int

class EmergencyContact(BaseModel):
    name: str
    relation: str
    phone: str

class Subject(BaseModel):
    subjectId: str
    screeningNumber: str
    medicalRecordNumber: str
    dateOfConsent: str
    dateOfEnrollment: str
    randomizationDate: str
    treatmentArm: str
    age: int
    dateOfBirth: str
    gender: str
    ethnicity: str
    heightCm: int
    weightKg: int
    BMI: float
    bloodType: str
    smokingStatus: str
    alcoholConsumption: str
    drugUse: str
    comorbidities: List[str]
    priorMedications: List[str]
    familyHistory: List[str]
    allergies: List[str]
    dietaryRestrictions: str
    geneticMarkers: str
    visitCount: int
    lastVisitDate: str
    nextScheduledVisit: str
    status: str
    finalStatus: str
    dateOfLastContact: str
    previousTrial: bool 
    progressStatus: str
    protocolDeviation: bool 
    discontinuationReason: Optional[str] = None
    dropoutReason: Optional[str] = None
    vaccinationStatus: bool 
    socioeconomicStatus: str
    incomeBracket: str
    educationLevel: str
    adherenceScore: float
    medicationAdherence: float
    complications: str
    vitalSigns: VitalSigns
    qualityOfLifeScores: float
    studyDrugDosage: str
    insuranceProvider: str
    emergencyContact: EmergencyContact
    
    siteId: Optional[str] = None
    siteName: Optional[str] = None
    protocolId: Optional[str] = None
    protocolName: Optional[str] = None
    companyId: Optional[str] = None
    companyName: Optional[str] = None

File: P32_gbeex_v4/gbeex-frontend/app/types/subjects.ts
// app/types/subject.ts

export interface VitalSigns {
    bloodPressure: string;
    heartRate: number;
}

export interface EmergencyContact {
    name: string;
    relation: string;
    phone: string;
}

export interface Subject {
    subjectId: string;
    screeningNumber: string;
    medicalRecordNumber: string;
    dateOfConsent: string;
    dateOfEnrollment: string;
    randomizationDate: string;
    treatmentArm: string;
    age: number;
    dateOfBirth: string;
    gender: string;
    ethnicity: string;
    heightCm: number;
    weightKg: number;
    BMI: number;
    bloodType: string;
    smokingStatus: string;
    alcoholConsumption: string;
    drugUse: string;
    comorbidities: string[];
    priorMedications: string[];
    familyHistory: string[];
    allergies: string[];
    dietaryRestrictions: string;
    geneticMarkers: string;
    visitCount: number;
    lastVisitDate: string;
    nextScheduledVisit: string;
    status: string;
    finalStatus: string;
    dateOfLastContact: string;
    previousTrial: boolean;
    progressStatus: string;
    protocolDeviation: boolean;
    discontinuationReason?: string;
    dropoutReason?: string;
    vaccinationStatus: boolean;
    socioeconomicStatus: string;
    incomeBracket: string;
    educationLevel: string;
    adherenceScore: number;
    medicationAdherence: number;
    complications: string;
    vitalSigns: VitalSigns;
    qualityOfLifeScores: number;
    studyDrugDosage: string;
    insuranceProvider: string;
    emergencyContact: EmergencyContact;

    siteId?: string;
    siteName?: string;
    protocolId?: string;
    protocolName?: string;
    companyId?: string;
    companyName?: string;
}